data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -17,14 +17,31 @@
  * This class implements the Sink interface and records boilerplate metrics
  */
 public abstract class AbstractSink<T extends Record<?>> implements Sink<T> {
+    protected static final int NUMBER_OF_RETRIES = 600;
     protected final PluginMetrics pluginMetrics;
     private final Counter recordsInCounter;
     private final Timer timeElapsedTimer;
+    private Thread retryThread;
 
     public AbstractSink(final PluginSetting pluginSetting) {
         this.pluginMetrics = PluginMetrics.fromPluginSetting(pluginSetting);
         recordsInCounter = pluginMetrics.counter(MetricNames.RECORDS_IN);
         timeElapsedTimer = pluginMetrics.timer(MetricNames.TIME_ELAPSED);
+        retryThread = null;
+    }
+
+    public abstract void doInitialize() throws Exception;
+
+    @Override
+    public void initialize() {
+        try {
+            doInitialize();
+        } catch (Exception e) {
+        }
+        if (!isReady()) {
+            retryThread = new Thread(new SinkThread(this, NUMBER_OF_RETRIES));
```

Comment:
Why would you retry in a separate thread vs retrying in the code right here?

Reply:
Because we want to check for multiple sinks in multiple pipelines in parallel.